cfme/markers/uncollect.py

```diff
@@ -127,16 +132,13 @@ def pytest_collection_modifyitems(session, config, items):
     with log_path.join('uncollected.log').open('w') as f:
         for item in items:
             # First filter out all items who have the uncollect mark
-            if item.get_marker('uncollect') or not uncollectif(item):
-                # if a uncollect marker has been added,
-                # give it priority for the explanation
-                uncollect = item.get_marker('uncollect')
-                marker = uncollect or item.get_marker('uncollectif')
-                if marker:
-                    reason = marker.kwargs.get('reason', "No reason given")
-                else:
-                    reason = None
-                f.write("{} - {}\n".format(item.name, reason))
+            uncollectif_result, uncollectif_reason = uncollectif(item)
```

Comment:
just wondering if it wouldnt make sense to just have a function that extracts the uncollect reason, and if its None, it doesnt get uncollected

that would simplify uncollect logging quite a bit and break the conditions down

Reply:
Later yes